### PR TITLE
docs(optimize): create migration guide for 3.12

### PR DESCRIPTION
**Pull request by misiekhardcore**
## When should this change go live?

- [x] This change is not yet live and should not be merged until 8.4 release date (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
+---
+id: 3.11_8.3-to-3.12_8.4
+title: "Update notes (8.3/3.11 to 8.4/3.12)"
+---
+
+:::note Heads up!
+To update Optimize to version 8.4/3.12, perform the steps in the [migration and update instructions](./instructions.md).
```
Comment (RomanJRW) [inline optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md:7]:
```suggestion
To update Optimize to version 3.12/8.4, perform the steps in the [migration and update instructions](./instructions.md).
```
Nitpick, but we generally do it this way round. Let's be consistent

File: optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+:::
+
+The update to 8.4/3.12 can be performed from any 8.3/3.11 release.
```
Comment (RomanJRW) [inline optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md:10]:
```suggestion
The update to 3.12/8.4 can be performed from any 3.11/8.3 release.
```

File: optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+- Known issues
+- Changes in supported environments
+- Changes in behavior (for example, due to a new feature)
+- Changes in translation resources
+
+## Changes in supported environments
+
+### Elasticsearch
+
+With 8.4/3.12, Optimize now supports Elasticsearch `8.9`.
```
Comment (RomanJRW) [inline optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md:24]: Everything below this line all the way down to the translations sections is not needed, it was only relevant for the update to 3.11 :)

File: optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+| --------------------------------------- | -------------------------------- |
+| Optimize 3.10.0 - Optimize 3.10.3       | 7.16.2+, 7.17.0+, 8.5.0+, 8.6.0+ |
+| Optimize 3.10.4                         | 7.16.2+, 7.17.0+, 8.7.0+, 8.8.0+ |
+| Optimize 3.10.5 - Optimize 8.3.x/3.11.x | 7.16.2+, 7.17.0+, 8.5.0+, 8.6.0+ |
+| Optimize 8.3.x/3.11.x                   | 8.8.0+                           |
+| Optimize 8.4.x/3.12.x                   | 8.9.0+                           |
+
+See the [supported environments]($docs$/reference/supported-environments) section for the full range of supported versions.
+
+If you need to update your Elasticsearch cluster, refer to the general [Elasticsearch update guide](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-upgrade.html). Usually, the only thing you need to do is perform a [rolling update](https://www.elastic.co/guide/en/elasticsearch/reference/current/rolling-upgrades.html).
```
Comment (RomanJRW) [inline optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md:37]: actually maybe keep these lines

File: optimize/self-managed/optimize-deployment/migration-update/instructions.md [hunk 5]
```diff
@@ -10,8 +10,8 @@ If you want to update Optimize by several versions, you cannot do that at once,
 
 | Update from  | Recommended update path to 8.3/3.11                               |
 | ------------ | ----------------------------------------------------------------- |
-| 8.3/3.11     | You are on the latest version.                                    |
-| 3.0 - 3.10.x | Rolling update to 8.3/3.11                                        |
+| 8.4/3.12     | You are on the latest version.                                    |
```
Comment (RomanJRW) [inline optimize/self-managed/optimize-deployment/migration-update/instructions.md:13]:
```suggestion
| 3.12/8.4     | You are on the latest version.                                    |
```

File: optimize/self-managed/optimize-deployment/migration-update/instructions.md [hunk 6]
```diff
@@ -10,8 +10,8 @@ If you want to update Optimize by several versions, you cannot do that at once,
...
+| 3.0 - 3.11.x | Rolling update to 8.4/3.12                                        |
```
Comment (RomanJRW) [inline optimize/self-managed/optimize-deployment/migration-update/instructions.md:14]:
```suggestion
| 3.0 - 3.11.x/8.3.x | Rolling update to 3.12/8.4                                        |
```

File: optimize/self-managed/optimize-deployment/migration-update/translation-diffs/differences_localization_311_312.diff [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
++    "details": "Details",
+     "process": {
+       "label": "Process",
+       "label-plural": "Processes"
+@@ -1285,6 +1273,7 @@
+         "duration": "Duration",
+         "instanceDuration": "Instance Duration",
+         "processInstanceDuration": "Process Instance Duration",
++        "dur": "Flow Node Duration",
+         "flowNodeDuration": "Flow Node Duration",
```
Comment (RomanJRW) [inline optimize/self-managed/optimize-deployment/migration-update/translation-diffs/differences_localization_311_312.diff:203]: is this right? Having two fields here with the same value feels wrong. Are they being used in FE?

File: optimize/self-managed/optimize-deployment/migration-update/translation-diffs/differences_localization_311_312.diff [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
++    "on": "On",
+     "off": "Off",
+     "none": "None",
+     "all": "All",
+@@ -1110,6 +1093,11 @@
+     "less": "Less",
+     "configure": "Configure",
+     "new": "New",
++    "error": "Error",
++    "closeError": "Close error notification",
```
Comment (RomanJRW) [inline optimize/self-managed/optimize-deployment/migration-update/translation-diffs/differences_localization_311_312.diff:191]: is this even used?

**Review by RomanJRW — CHANGES_REQUESTED** [review]
Thanks for putting this together 💪 
Some comments from me, feel free to check the `3.12/8.4` vs `8.4/3.12` thing with @christinaausley too, she will have a good answer for this I'm sure 🙂

File: optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md (hunk 4 above)
Comment (christinaausley) [inline optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md:37]: I worry about using a term like "usually" -- if there is another instance where I might need to do something else, please clarify 👍 If a rolling update is all I will need 99% of the time, we can remove the "usually" here.

File: optimize/self-managed/optimize-deployment/migration-update/translation-diffs/differences_localization_311_312.diff
Comment (christinaausley) [inline optimize/self-managed/optimize-deployment/migration-update/translation-diffs/differences_localization_311_312.diff:1]: This looks inline with work I did in https://github.com/camunda/camunda-optimize/pull/9558.

Am I clear to review these lines as well according to the WCoE (mostly minor things, like sentence-case).

**Review by christinaausley — COMMENTED** [review]
I think this just needs to be added to the sidebar 👍

File: optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md (hunk 1 above)
Comment (RomanJRW) [reply optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md:7]: actually I might have been wrong about this, let's leave it

File: optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md (hunk 2 above)
Comment (RomanJRW) [reply optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md:10]: as above

File: optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md (hunk 4 above)
Comment (RomanJRW) [reply optimize/self-managed/optimize-deployment/migration-update/3.11_8.3-to-3.12_8.4.md:37]: On major version changes, rolling upgrades are often not supported by ES. But it's very much out of our hands. Happy to change to whatever you think, please advise as to what you think is best fit 🙂

File: optimize/self-managed/optimize-deployment/migration-update/translation-diffs/differences_localization_311_312.diff
Comment (RomanJRW) [reply optimize/self-managed/optimize-deployment/migration-update/translation-diffs/differences_localization_311_312.diff:1]: Good point. I will ask @OmranAbazid to merge the linked PR

@misiekhardcore - once this is done, can you please add the changed file to this PR before merging?

**Comment by RomanJRW** [comment]
I think there was lack of clarity around 8.8 support, so I have pushed some changes to align things and make some of the review changes (@misiekhardcore is not working today). I think the remaining tasks for this are:

- [ ] @OmranAbazid gets https://github.com/camunda/camunda-optimize/pull/9558 merged
- [ ] @misiekhardcore adds the updated translation file to the published diff, replacing the one in this PR
- [ ] @christinaausley re-reviews and merges if appropriate

Please let me know if you need any support from my end. I think a realistic target for this would be on Monday next week

**Comment by christinaausley** [comment]
Works for me! Sorry for my crossover changes here.

File: optimize/self-managed/optimize-deployment/migration-update/translation-diffs/differences_localization_311_312.diff (hunk 8 above)
Comment (misiekhardcore) [reply optimize/self-managed/optimize-deployment/migration-update/translation-diffs/differences_localization_311_312.diff:191]: yes, this is used

File: optimize/self-managed/optimize-deployment/migration-update/translation-diffs/differences_localization_311_312.diff (hunk 7 above)
Comment (misiekhardcore) [reply optimize/self-managed/optimize-deployment/migration-update/translation-diffs/differences_localization_311_312.diff]: I believe this is used by the filter type

File: optimize/self-managed/optimize-deployment/migration-update/translation-diffs/differences_localization_311_312.diff
Comment (misiekhardcore) [reply optimize/self-managed/optimize-deployment/migration-update/translation-diffs/differences_localization_311_312.diff:1]: translation diff is now updated

**Comment by RomanJRW** [comment]
@christinaausley - just pinging as a heads up, I think this is ready for review on your end now. Let us know if you need anything 🙂

**Review by RomanJRW — APPROVED** [review]
LGTM

**Review by christinaausley — APPROVED** [review]

**Comment by christinaausley** [comment]
This is on `hold` until the `8.4` release, yes?

**Comment by misiekhardcore** [comment]
> This is on `hold` until the `8.4` release, yes?

@christinaausley yes

**Comment by misiekhardcore** [comment]
@christinaausley I had to add few changes to this PR

**Review by christinaausley — APPROVED** [review]

**Comment by christinaausley** [comment]
@misiekhardcore Are you comfortable merging this in for release next week?

**Comment by misiekhardcore** [comment]
@christinaausley yes

**Pull request merged**